src/stream/src/common/log_store/kv_log_store/writer.rs

```diff
@@ -107,6 +113,16 @@ impl<LS: LocalStateStore> LogWriter for KvLogStoreWriter<LS> {
                 let (key, value) = self.serde.serialize_barrier(epoch, vnode, is_checkpoint);
                 self.state_store.insert(key, value, None)?;
             }
+            self.tx
+                .flush_all_unflushed(|chunk, epoch, start_seq_id, end_seq_id| {
+                    for (i, (op, row)) in chunk.rows().enumerate() {
+                        let seq_id = start_seq_id + (i as SeqIdType);
+                        assert!(seq_id <= end_seq_id);
+                        let (_, key, value) = self.serde.serialize_data_row(epoch, seq_id, op, row);
+                        self.state_store.insert(key, value, None)?;
```

Comment:
In L95, we will explicitly trigger one flush per chunk but here we wait until L136 to trigger a flush. Shall we make it consistent (though I don't see any correctness issue here)?

Reply:
We will construct one imm per chunk in `write_chunk` (L95). I think it will also generate many small imms?

Reply:
Yes. But in `write_chunk` we have no choice to do it so that log reader can read from storage right after `write_chunk` finishes. Here we don't have to do it.